world.js

```diff
@@ -120,6 +120,12 @@ var createWorldCreator = function() {
                     }
                 });
             });
+            elevator.on("indicatorstate_change", function() {
+                var elevator = this;
+                if(elevator.isOnAFloor() && !elevator.isMoving && !elevator.isFull()) {
```

Comment:
I did think about that, but I wanted to touch as little existing code as possible.  I could certainly pull that after this gets merged.